src/query/api/v1/handler/json/write.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package json
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/m3db/m3/src/query/api/v1/handler"
+	"github.com/m3db/m3/src/query/storage"
+	"github.com/m3db/m3/src/query/ts"
+	"github.com/m3db/m3/src/query/util/logging"
+	xtime "github.com/m3db/m3x/time"
+
+	"go.uber.org/zap"
+)
+
+const (
+	// WriteJSONURL is the url for the write json handler
+	WriteJSONURL = handler.RoutePrefixV1 + "/write_json"
+
+	// JSONWriteHTTPMethod is the HTTP method used with this resource.
+	JSONWriteHTTPMethod = http.MethodPost
+)
+
+// WriteJSONHandler represents a handler for the write json endpoint
+type WriteJSONHandler struct {
+	store storage.Storage
+}
+
+// NewWriteJSONHandler returns a new instance of handler.
+func NewWriteJSONHandler(store storage.Storage) http.Handler {
+	return &WriteJSONHandler{
+		store: store,
+	}
+}
+
+// WriteQuery represents the write request from the user
+// NB(braskin): support only writing one datapoint for now
+type WriteQuery struct {
+	Tags      map[string]string `json:"tags" validate:"nonzero"`
+	Timestamp int               `json:"timestamp" validate:"nonzero"`
```

Comment:
Is this fine or should I make it take a `ts.Datapoint`?

Reply:
Instead of Timestamp int, can you use the same thing we use in /prometheus/native/read endpoint ? We have parseTime in m3/src/query/api/v1/handler/prometheus/native/common.go which you can probably take out

Reply:
So the way it is in m3/src/query/api/v1/handler/prometheus/native/common.go, would that mean we would need to pass in the write request info in the url parameters as opposed to the body?

Reply:
I think you can just make `Timestamp` a string, and use `github.com/m3db/m3/src/query/util.ParseTimeString()` to read it to a time.Time